### ISSUE TITLE
Compile fails in pl-coverage.c with some compilers

### ISSUE THREAD

**Issue by md5i** [SWI-Prolog/swipl-devel#988]
At src/pl-coverage.c:176 some C compilers will emit an error because you have a declaration immediately following a label.  Example output:
```
swipl/src/pl-coverage.c:176:7: error: a label can only be part of a statement and a declaration is not a statement
  176 |       ClauseRef cref = clause_clref(cov->cref);
      |       ^~~~~~~~~
```
See https://stackoverflow.com/questions/18496282/why-do-i-get-a-label-can-only-be-part-of-a-statement-and-a-declaration-is-not-a for the reason why.  Simple solutions: Put the case branch in a block `{ ... }` or make an empty statement after the label `;`.

**Comment by JanWielemaker** [comment]
This issue has been mentioned on **SWI-Prolog**. There might be relevant details there:

https://swi-prolog.discourse.group/t/mac-m1/4722/14

**Comment by md5i** [comment]
Yes, that is one of the two solutions I mentioned.  I would make a PR, but I am not allowed to sign the disclaimer that it requires.
